File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k=0;
        for(int j=1;j<nums.length;j++){
            if(nums[j]!=nums[k]){
                k++;
                nums[k]=nums[j];
            }
        }

        k=k+1;
        return k;


        // Map<Integer,Integer> map = new HashMap<>();
        // int k=0;
        // for(int i=0;i<nums.length;i++){
        //     if(!map.containsKey(nums[i])){
        //         // swap(nums[k],nums[i])
        //         int temp = nums[k];
        //         nums[k] = nums[i];
        //         nums[i] = nums[k];

        //         map.put(nums[i],i);
        //         k++;
        //     }
        // }
        // return k;
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // set<int> st;
        // for(int i=0;i<nums.size();i++){
        //     st.insert(nums[i]);
        // }
        // int ind=0;
        // for(auto it:st){
        //     nums[ind] = it;
        //     ind++;
        // }

        // return ind;

        int i=0,j=1;
        
        while(j<nums.size()){
            if(nums[j]!=nums[i]){
                i++;
                nums[i] = nums[j];
            }
            j++;
        }

        return i+1;
    }

    //     int j = 1;
    //     for(int i = 1; i < nums.size(); i++){
    //         if(nums[i] != nums[i - 1]){//i=1,j=1|i=2,j=1
    //             nums[j] = nums[i];//nums[1] = 2
    //             j++;//j=2
    //         }
    //     }
    //     return j;
    // }
};